### PR TITLE
[OSD-19695] make sure the binaries are up to date

### PR DESCRIPTION
**Pull request by bmeng**
### What type of PR is this?

_feature_

### What this PR does / Why we need it?

Make the binaries in the image are up to date

### Which Jira/Github issue(s) does this PR fix?

_Resolves #[OSD-19695](https://issues.redhat.com//browse/OSD-19695)_

### Pre-checks (if applicable)

- [ ] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -90,12 +94,21 @@ WORKDIR /ocm-cli
...
 
 # Checking the SHA-256 hash
-RUN echo $OCM_SHA256_CHECKSUM | sha256sum -c -
+RUN curl -sSLf $OCM_SHA256_CHECKSUM_URL | sha256sum -c -
 
 # Moving the validated ocm binary to /out and make it executable
 RUN mv ocm-linux-amd64 /out/ocm
 RUN chmod +x /out/ocm
 
+# Install golang with specific version
```
Comment (xiaoyu74) [inline Dockerfile:103]: I built a docker image based on your updated Dockerfile and run a container
```
docker run -it shawn-test /bin/bash
[root@cdb748e8bb57 ~]#
[root@cdb748e8bb57 ~]# go version
bash: go: command not found

echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Looks  no `go path` included.  Can you double check? 

The `ocm version` and `yq --version` look all good.

Comment (bmeng) [reply Dockerfile:103]: The go installed in the builder only, for building the hypershift, and it will copy the hypershift binary only to the final image

Comment (xiaoyu74) [reply Dockerfile:103]: oh, I see. Thanks. 
I can see the Acceptance Criteria also mentions: `An automated dependency upgrade system for binaries set up in the managed scripts repository.`  If it's an optional and not considering in this PR, then LGTM.

Comment (bmeng) [reply Dockerfile:103]: As long as all the binaries/packages can be up to date all the time. I think it is kind of covered.

**Pull request merged**